jake/pip/pip.py

```diff
@@ -24,19 +24,25 @@
 
 class Pip():
   """pip.py gets installed pip dependencies"""
-  def __init__(self, targets=None):
+  def __init__(self, targets=None, requirements_file_path=None):
     self._log = logging.getLogger('jake')
     self._format = "pypi"
     self._working_set = pkg_resources.working_set
+    self.requirements_file_path = requirements_file_path
     if targets:
       self._working_set = pkg_resources.WorkingSet(ast.literal_eval(targets))
-    self._coords = self.generate_dependencies()
+    self.__coords = self.generate_dependencies()
 
   def generate_dependencies(self, coords=Coordinates()) -> (Coordinates):
     """converts list of pkg_resource.working_set into purl coordinates"""
 
     for i in iter(self._working_set):
       coords.add_coordinate(i.project_name, i._version, self._format)
+    if self.requirements_file_path:
+      with open(self.requirements_file_path, 'r') as requirements_file:
+        for line in requirements_file.readlines():
+          name, version = line.strip().split('==')
+          coords.add_coordinate(name, version, self._format)
```

Comment:
Let me start off by saying I'm not a python expert. :) Feel free to correct me as needed.

I'm scratching my head about how adding the `requirements` file components to the discovered components (`coords`) is intended to work. When I read the README.md description of the new `-r` feature, I get the impression the desire is to scan the components listed in a `requirements` file INSTEAD of the components Jake "self-discovers" via the `WorkingSet`. 

If I read this code correctly, it appears  the components in the `requirements` file will be ADDED to the list of components  Jake "self-discovers" via the `WorkingSet`.  (This results in `coords` including BOTH the components Jake "self-discovers" AND those in the `requirements` file). Is that the desired behavior?

It might be worth changing either the code, or the README.md to clarify the behavior/intent.

Reply:
Is there any auto discovering?
If I've understood the code correctly, jake receives the dependencies either as `targets` flag or from output of the `conda list` command. I tries to add a third way which is receiving a requirements file path and reading the dependencies from that. So I think the README is already clarifying.

Reply:
Got it now. I'll fix.